sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -140,9 +146,18 @@ def on_request(self, request):
         :param request: The PipelineRequest object
         :type request: ~azure.core.pipeline.PipelineRequest
         """
-        request_id = request.context.options.pop('request_id', self._request_id)
-        if request_id:
-            header = {"x-ms-client-request-id":request_id}
+        update_header = False
+        if 'request_id' in request.context.options:
+            request_id = request.context.options.pop('request_id')
+            update_header = True
+        elif self._request_id is not _Unset:
+            request_id = self._request_id
+            update_header = True
+        elif self._auto_request_id:
+            request_id = str(uuid.uuid1())
+            update_header = True
```

Comment:
Why do you need this boolean? You can init a "request_id" variable to _Unset, and if at the end of all if it's still _Unset, don't send anything.